src/htr2hpc/train/run.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python
+import argparse
+import os
+import sys
+import logging
+import pathlib
+import time
+
+from intspan import intspan
+from kraken.kraken import SEGMENTATION_DEFAULT_MODEL, DEFAULT_MODEL
+from tqdm import tqdm
+
+from htr2hpc.api_client import eScriptoriumAPIClient
+from htr2hpc.train.apps import (
+    get_training_data,
+    segtrain,
+    get_model,
+    slurm_job_status,
+    slurm_job_queue_status,
+)
+
+
+api_token_env_var = "ESCRIPTORIUM_API_TOKEN"
+
+# map our job type option choices to escriptorium terms
+es_model_jobs = {"segmentation": "Segment", "transcription": "Recognize"}
+
+# kraken python package provides paths to the default best models for both modes
+default_model = {
+    "segmentation": SEGMENTATION_DEFAULT_MODEL,
+    "transcription": DEFAULT_MODEL,
+}
+
+
+def main():
+    try:
+        api_token = os.environ[api_token_env_var]
+    except KeyError:
+        print(
+            f"Error: eScriptorium API token must be set as environment variable {api_token_env_var}",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    # TODO: add options for:
+    # - create/update model flag (make explicit)
+    #   name for the new model when creating a new one (required)
+
+    # use subparsers for the two modes
+    parser = argparse.ArgumentParser(
+        description="Export content from eScriptorium and train or fine-tune models"
+    )
+    subparsers = parser.add_subparsers(
+        title="mode", description="supported training modes", required=True, dest="mode"
+    )
+    subparsers.add_parser("segmentation")  # currently no segmentation-specific options
+    transcription_parser = subparsers.add_parser("transcription")
+
+    # common arguments used in both modes
+    parser.add_argument(
+        "base_url",
+        metavar="BASE_URL",
+        help="Base URL for eScriptorium instance (without /api/)",
+        type=str,
+    )
+    parser.add_argument(
+        "work_dir",
+        metavar="WORKING_DIR",
+        help="Working directory where data should be downloaded (must already exist)",
+        type=pathlib.Path,
+    )
+    parser.add_argument(
+        "-d",
+        "--document",
+        help="Document id to export",
+        type=int,
+        dest="document_id",
+        required=True,
+    )
+    parser.add_argument(
+        "-m",
+        "--model",
+        help="Optional model id to use for fine-tuning",
+        type=int,
+        dest="model_id",
+    )
+
```

Comment:
We probably will want to pass the name for the new model through the command, as this will be something the user would input when submitting a training job.

```suggestion
    parser.add_argument(
        "--model-name",
        help="Name to be used for the newly trained model",
        type=str,
        dest="model_name",
        required=True,
    )
```